packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -193,13 +193,18 @@ export const useField = <
       return;
     }
 
-    const valueStr = event.target.value;
+    const targetValue = event.target.value;
+    const eventData = (event.nativeEvent as InputEvent).data;
+    // Calling `.fill(04/11/2022)` in playwright will trigger a change event with the requested content to insert in `event.nativeEvent.data`
+    // usual changes have only the currently typed character in the `event.nativeEvent.data`
+    const useEventData = eventData && eventData.length > 1;
```

Comment:
Small nitpick

```suggestion
    const shouldUseEventData = eventData && eventData.length > 1;
```

To avoid having a variable with a hook-like name.